tsl/src/dist_util.c

```diff
@@ -357,15 +353,26 @@ dist_util_is_compatible_version(const char *data_node_version, const char *acces
 				(errcode(ERRCODE_INTERNAL_ERROR),
 				 errmsg("invalid access node version %s", access_node_version)));
 
+	/* if major versions are same, compare minor versions */
 	if (data_node_major == access_node_major)
-		if (data_node_minor == access_node_minor)
-			*is_old_version = (data_node_patch < access_node_patch);
+	{
+		/*
+		 * if minor versions are same (or newer on DN), we ignore patch versions and
+		 * declare compatibility. If minor version of DN is lower then it's not
+		 * compatible
+		 */
+		if (data_node_minor >= access_node_minor)
+			is_compatible = true;
 		else
-			*is_old_version = (data_node_minor < access_node_minor);
+			is_compatible = false;
+	}
+	/* if DN major version is newer, then still compatible. Else not */
+	else if (data_node_major > access_node_major)
```

Comment:
Nit; you can just return the boolean value, no need to save it away in a variable that you then return.
```
else
  return (data_node_major > access_node_major);
```

Reply:
ok, done